osu.Game/Input/OsuConfineMouseMode.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.ComponentModel;
+using osu.Framework.Input;
+
+namespace osu.Game.Input
+{
+    /// <summary>
+    /// Determines the situations in which the mouse cursor should be confined to the window.
+    /// Expands upon <see cref="ConfineMouseMode"/> by providing the option to confine during gameplay.
+    /// </summary>
+    public enum OsuConfineMouseMode
+    {
+        /// <summary>
+        /// The mouse cursor will be free to move outside the game window.
+        /// </summary>
+        Never,
+
+        /// <summary>
+        /// The mouse cursor will be locked to the window bounds while in fullscreen mode.
+        /// </summary>
+        Fullscreen,
+
+        /// <summary>
+        /// The mouse cursor will be locked to the window bounds during gameplay,
+        /// but may otherwise move freely.
+        /// </summary>
+        [Description("During Gameplay")]
+        DuringGameplay,
```

Comment:
Rename to `WhenOverlaysDisabled` for clarity? I think it should work better.

Reply:
Leave the `Description` as is I assume?

Reply:
yep